resolver/resolver.go

```diff
@@ -18,12 +18,34 @@ import (
 	"context"
 	"io"
 	"net"
+	"net/netip"
 	"time"
 
 	"github.com/bufbuild/httplb/attribute"
 	"github.com/bufbuild/httplb/internal"
 )
 
+// AddressFamilyAffinity is an option that allows control over the preference
+// for which addresses to consider when resolving, based on their address
+// family.
+type AddressFamilyAffinity int
+
+const (
+	// AllFamilies will result in all addresses being used, regardless of
+	// their address family.
+	AllFamilies AddressFamilyAffinity = iota
```

Comment:
Thinking about this more, I think it's a little odd for this choice to be the zero value. Maybe instead we order them (incorporating idea from other comment):
* `PreferIPv4`
* `RequireIPv4`
* `PreferIPv6`
* `RequireIPv6`
* `UseBothIPv4AndIPv6`

I changed `AllFamilies` to `UseBothIPv4AndIPv6` in the example list above since I think that would make call sites a littler more readable as to what that argument means. In that same vein, maybe rename the type to `AddressFamilyPolicy`?

WDYT?

Reply:
Hmmm, sure, that all sounds good. It seems like this makes things a little clearer/simpler from the user perspective in exchange for slightly overloading what we actually do on our end.

Reply:
Done.